backup.diff

```diff
@@ -0,0 +1,145 @@
+@@ -63,7 +63,10 @@ public class GracefulRollingRestartRsAction extends RestartActionBaseAction {
+         rm.load();
+       } catch (Shell.ExitCodeException e) {
+         getLogger().info("Problem restarting but presume successful; code={}", e.getExitCode(), e);
++      } catch (Exception e) {
++        getLogger().info("Got Exception: " + e + ", continue");
+       }
++
+       sleep(RandomUtils.nextInt(0, (int)sleepTime));
+     }
+     getLogger().info("Enabling balancer");
+diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActionBaseAction.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActionBaseAction.java
+index ce41522fe7..d927055a3f 100644
+--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActionBaseAction.java
++++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActionBaseAction.java
+@@ -68,6 +68,8 @@ public abstract class RestartActionBaseAction extends Action {
+     sleep(sleepTime);
+     getLogger().info("Starting region server: {}", server);
+     startRs(server);
++    // Sleep 2 seconds to make sure RS is online.
++    sleep(2000);
+   }
+ 
+   void restartRs(ServerName server, long sleepTime) throws IOException {
+diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/ServerAndDependenciesKillingMonkeyFactory.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/ServerAndDependenciesKillingMonkeyFactory.java
+index 676b04d3fe..024d3d4e21 100644
+--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/ServerAndDependenciesKillingMonkeyFactory.java
++++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/ServerAndDependenciesKillingMonkeyFactory.java
+@@ -50,16 +50,16 @@ public class ServerAndDependenciesKillingMonkeyFactory extends MonkeyFactory {
+ 
+     // Destructive actions to mess things around. Cannot run batch restart.
+     Action[] actions1 = new Action[]{
+-      new RestartRandomRsExceptMetaAction(60000),
+-      new RestartActiveMasterAction(5000),
+-      new RollingBatchRestartRsAction(5000, 1.0f, 2,
+-        true), // only allow 2 servers to be dead.
+-      new ForceBalancerAction(),
+-      new RestartRandomDataNodeAction(60000),
+-      new RestartRandomZKNodeAction(60000),
++      //new RestartRandomRsExceptMetaAction(60000),
++      //new RestartActiveMasterAction(5000),
++      //new RollingBatchRestartRsAction(5000, 1.0f, 2,
++      //  true), // only allow 2 servers to be dead.
++      //new ForceBalancerAction(),
++      //new RestartRandomDataNodeAction(60000),
++      //new RestartRandomZKNodeAction(60000),
+       new GracefulRollingRestartRsAction(gracefulRollingRestartTSSLeepTime),
+-      new RollingBatchSuspendResumeRsAction(rollingBatchSuspendRSSleepTime,
+-          rollingBatchSuspendtRSRatio)
++      //new RollingBatchSuspendResumeRsAction(rollingBatchSuspendRSSleepTime,
++      //    rollingBatchSuspendtRSRatio)
```

Comment:
I guess these were left commented by mistake? Otherwise, could you explain why none of these actions are needed anymore and remove it altogether from this code?

Reply:
Sorry, by accident I uploaded a debug version, let me correct it.